extensions/amp-tiktok/0.1/test/test-amp-tiktok.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '../amp-tiktok';
+import {createElementWithAttributes, isAmpElement} from '../../../../src/dom';
+
+describes.realWin(
+  'amp-tiktok',
+  {
+    amp: {
+      extensions: ['amp-tiktok'],
+    },
+  },
+  (env) => {
+    let win;
+    let doc;
+    let clock;
+
+    beforeEach(() => {
+      win = env.win;
+      doc = win.document;
+      clock = env.sandbox.useFakeTimers();
+      clock.tick(0);
+    });
+
+    async function getTiktok(attrs = {}) {
+      const tiktok = createElementWithAttributes(win.document, 'amp-tiktok', {
+        layout: 'responsive',
+        width: '325px',
+        height: '730px',
+        ...attrs,
+      });
+      // if (isFullSrcUrl) {
+      //   tiktok.setAttribute(
+      //     'data-src',
+      //     `https://www.tiktok.com/@scout2015/video/${videoId}?lang=en-US`
+      //   );
+      // } else if (opt_locale) {
+      //   tiktok.setAttribute('data-locale', opt_locale);
+      //   tiktok.setAttribute('data-src', videoId);
+      // } else {
+      //   tiktok.setAttribute('data-src', videoId);
+      // }
+      doc.body.appendChild(tiktok);
+      return tiktok
+        .buildInternal()
+        .then(() => {
+          return tiktok.layoutCallback();
+        })
+        .then(() => tiktok);
+    }
+
+    it('renders with full src url', async () => {
+      const videoId = '6718335390845095173';
+      const player = await getTiktok({src: videoId});
```

Comment:
(Since I can't respond to https://github.com/ampproject/amphtml/pull/32651#discussion_r591652854 so starting a new thread)

We shouldn't be depending on external content in a unit test. If Tiktok goes down or this video is deleted, our tests will fail. As Alan recommended, load `blank.html` from `localhost`. I understand this is done in other 3p iframe extensions, but Tiktok can be a start and not add to tech debt 🙂

Reply:
In this case I am trying to verify that the code which verifies and creates the src url executes correctly in buildcallback, do you think this would be better tested in an e2e context?

Reply:
Yes, an e2e test would be better. I'd still recommend trying to load blank.html from localhost in the e2e test though.

Reply:
I will be adding an e2e test in a folowing PR, updated this test as reccomended.